apps/wpcom-block-editor/src/calypso/features/iframe-bridge-server.js

```diff
@@ -604,10 +607,15 @@ function openLinksInParentFrame() {
 		'.components-snackbar-list .components-snackbar__content a', // View Post link in success snackbar, Gutenberg >=5.9
 		'.post-publish-panel__postpublish .components-panel__body.is-opened a', // Post title link in publish panel
 		'.components-panel__body.is-opened .post-publish-panel__postpublish-buttons a.components-button', // View Post button in publish panel
+		// '.components-snackbar .components-snackbar__content a.components-button', // .components-snackbar__action.is-tertiary
 	].join( ',' );
 	$( '#editor' ).on( 'click', viewPostLinkSelectors, ( e ) => {
 		e.preventDefault();
-		window.open( e.target.href, '_top' );
+		if ( isDesktop ) {
```

Comment:
As far as I can tell, this seems to be the correct place to override this behavior (i.e. disable the `window.open` and instead send the desktop app a custom event. However: (1) the default behavior is occurring anyway, and (2) the desktop event is not being sent. 🤔

Reply:
👋 Pinging @mmtr @kwight as I see you might be familiar with this logic in the iframe bridge server, or could perhaps help point me in the right direction?

Reply:
@nsakaimbo similarly to FSE, the `wpcom-block-editor` app doesn't run in Calypso. Did you generate a new build and uploaded it to your WP.com sandbox? See PCYsg-l4k-p2.

Also, since this file runs in the WP Admin context and not in Calypso, dispatching a Calypso action here won't have any any effect. In order to communicate with Calypso, you need to send a message through the iframe port communication channel ([see example](https://github.com/Automattic/wp-calypso/blob/7b12ea2ddd42b82852726fd7f452b8cc88cccee3/apps/wpcom-block-editor/src/calypso/features/iframe-bridge-server.js#L476-L484)) and then listen for that message in Calypso so you can dispatch the desired action ([see example](https://github.com/Automattic/wp-calypso/blob/7b12ea2ddd42b82852726fd7f452b8cc88cccee3/client/gutenberg/editor/calypsoify-iframe.tsx#L282-L285)).

Reply:
Thanks, @mmtr! To-date, I haven't had reason to use the WP.com sandbox, so this wasn't on my radar. Thanks for the links - will set up a sandbox and proceed accordingly.

To clarify: this means that in order for these changes to take effect, the final code changes must be deployed to both WP.com and Calypso?

Since this file runs in the WP Admin context, will checking for `config.isEnabled( 'desktop` )` within this file work? If not, I'm not sure how to proceed here since the default action (`window.open`) cannot be short-circuited based on environment/context.

Reply:
Let me know if you need any help @nsakaimbo!
> this means that in order for these changes to take effect, the final code changes must be deployed to both WP.com and Calypso?

Correct!

> Since this file runs in the WP Admin context, will checking for config.isEnabled( 'desktop ) within this file work?

Nope, you'd need to perform this check in the client listener. Something like:

`iframe-bridge-server.js`
```
calypsoPort.postMessage( 
  { 
    action: 'viewPost',
    payload: {
        postUrl: postUrl,
    },
  },
);
```

`calypsoify-iframe.tsx`
```
if ( 'viewPost' === action ) {
  const { postUrl } = payload;
  config.isEnabled( 'desktop' )
    ? dispatch( notifyDesktopViewPostClicked( postUrl );
    : window.location.href = postUrl;
}
config.isEnabled( 'desktop ) ?
```

Reply:
_Update_: the changes you suggested made sense and I went ahead and incorporated them in 4312a43ae1a0366d538eec26cb4774e119e3f4f2. This PR pairs with the changes in the desktop repo [here](https://github.com/Automattic/wp-desktop/compare/fix-view-post-handler).

However I haven't had much luck testing as I've been struggling to get the wp.com sandbox set up (I'm a `wp.com` sandbox noob 😅 ). I was able to follow the instructions to build the block editor and sync with `unison`, however am unclear how to point the desktop app to use the sandbox instead of production.

Reply:
Update: Made some progress and I think I'm most of the way towards sandboxing the block editor correctly: running Calypso from localhost while sandboxing the API as well as `widgets.wp.com`. However the changes don't appear to be working, so I might still be missing something.